src/seedu/addressbook/commands/employee/EmployeeEditCommand.java

```diff
@@ -0,0 +1,204 @@
+import seedu.addressbook.data.employee.UniqueEmployeeList.EmployeeNotFoundException;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+import seedu.addressbook.common.Messages;
+
+/**
+ * Edits the details of an existing employee in the Rms.
+ */
+public class EmployeeEditCommand extends Command {
+    public static final String COMMAND_WORD = "editemp";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the employee identified "
+            + "by the index number used in the displayed person list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[p/PHONE] "
+            + "[e/EMAIL] "
+            + "[a/ADDRESS] "
+            + "[pos/POSITION]\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + "p/97654321 ";
+
+    public static final String MESSAGE_EDIT_EMPLOYEE_SUCCESS = "Edited Person: %1$s";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+    public static final String MESSAGE_DUPLICATE_EMPLOYEE = "This person already exists in the address book.";
+
+    private final EditEmployeeDescriptor editEmployeeDescriptor;
+
+    /**
+     * @param targetVisibleIndex of the person in the filtered person list to edit
+     * @value editEmployeeDescriptor details to edit the employee with
+     */
+    public EmployeeEditCommand(int targetVisibleIndex,
+                               String phone,
+                               String email,
+                               String address,
+                               String position) throws IllegalValueException{
+        super(targetVisibleIndex);
+        this.editEmployeeDescriptor = new EditEmployeeDescriptor(phone, email, address, position);
+    }
+
+    // check for any non null parameters should already be done, probably in parser
+    @Override
+    public CommandResult execute() {
+        try {
+            final ReadOnlyEmployee employeeToEdit = getTargetEmployee();
+            Employee editedEmployee = createEditedEmployee(employeeToEdit, editEmployeeDescriptor);
+
+            rms.editEmployee(employeeToEdit, editedEmployee);
+
+            return new CommandResult(String.format(MESSAGE_EDIT_EMPLOYEE_SUCCESS, editedEmployee));
+
+        } catch (IndexOutOfBoundsException ie) {
+            return new CommandResult(Messages.MESSAGE_INVALID_EMPLOYEE_DISPLAYED_INDEX);
+        } catch (EmployeeNotFoundException enfe) {
+            return new CommandResult(Messages.MESSAGE_EMPLOYEE_NOT_IN_RMS);
+        }
+
+    }
+
+    /**
+     * Creates and returns an {@code Employee} with the details of {@code employeeToEdit}
+     * edited with {@code editEmployeeDescriptor}.
+     */
+    private static Employee createEditedEmployee(ReadOnlyEmployee employeeToEdit, EditEmployeeDescriptor editEmployeeDescriptor) {
```

Comment:
These methods should be put in `Rms.java`, the `Command` class should only contain its relevant constants, its own constructor and the `execute()` method

Reply:
Is it fine if this createEditedEmployee is in its own EditEmployeeDescriptor class?

Reply:
Yes, it is fine